src/olympia/devhub/tests/test_forms.py

```diff
@@ -1606,8 +1606,9 @@ def test_clean_adds_version(self):
             'version': uv1,
         }
 
-        # and when we select listed but there is no listed version availble: error
+        # and when we select listed but there is no listed version available: error
         data['channel'] = amo.CHANNEL_LISTED
+        del data['unlisted_version']
```

Comment:
Can we split this test into several separate ones instead ? There are 3 separate sets of assertions, we should probably just have 3 methods to reduce the chance of them affecting each other.